### DISCUSSION TITLE
Request for assistance: updating CLI to watchexec 2.x, tests to hyper 0.14.x

### DISCUSSION THREAD

**Discussion by pkgw** [tectonic-typesetting/tectonic#1017] [Announcements]
Hi folks!

You might have noticed that the Tectonic repo has been quiet lately. I've been putting in a lot of work on a cool project (*ooooo*) that I hope to launch soon; I just merged an enabling pull request here in the main repository in #1016.

I'd like to put out some new releases of the Tectonic crates that feature this new work, and before I do so, it would be pretty nice to update some of our dependencies:

- the main CLI depends on the 1.x version series of [watchexec], which includes old versions of some inner dependencies that lead to build warnings. Ideally we should migrate to the watchexec 2.x series
- the test suite uses the 0.12.x series of [hyper], which has associated security alerts, and duplicates the main crate which uses the 0.14.x series. This isn't a worry in practice since this code is only used for the tests, but it would be nice to update it.

[watchexec]: https://watchexec.github.io/
[hyper]: https://hyper.rs/

I started looking at the watchexec update today and it looks like it is not exactly straightforward to migrate to the new version. The update to hyper might also require some futzing around, although I haven't looked into it yet. If anyone with Rust development skills would be able to help with these projects, that would be really great!

**Comment by pkgw** [comment]
Oh, and to suppress another build warning, we should also update [quick-xml] in the main driver from the 0.22 series to 0.28. This may be a trivial change — I haven't looked into it either.

[quick-xml]: https://crates.io/crates/quick-xml

**Comment by CraftSpider** [comment]
I've started work on updating all three of these - will make a PR once it's done and passes all the tests.

**Reply by CraftSpider** [reply]
Opened a PR: #1033

**Reply by pkgw** [reply]
Thanks for all your work on this project, @CraftSpider ! Dealing with `watchexec` has looked like a lot more of a slog than we'd have hoped.
